FactSet/factset_fund_data.R

```diff
@@ -0,0 +1,198 @@
+data %>%
+  mutate(factset_fund_id_exists = !is.na(factset_fund_id)) %>%
+  group_by(factset_fund_id_exists) %>%
+  summarize(isin_count = n())
+# of the 1000 randomly pulled ISIN funds, 59 are missing altogether from the
+# FactSet dataset (around 94% coverage, by ISIN)
+
+sampled_factset_fund_ids <- unique(data$factset_fund_id)
+
+# `own_fund_detail` contains the holding details of each fund, by
+# `factset_fund_id`. Here, we will pull all relevant funds
+own_fund_detail <- tbl(factset_db, "own_v5_own_fund_detail") %>%
+  filter(factset_fund_id %in% sampled_factset_fund_ids) %>%
+  filter(report_date == data_timestamp) %>%
+  select(factset_fund_id, fsym_id, reported_mv) %>%
+  collect()
+
+# connect all the factset funds (where available) to details containing:
+# - fsym_id associated with each holding in fund
+# - reported market value associated with each holding
+data <- data %>%
+  select(fund_isin, factset_fund_id) %>%
+  left_join(own_fund_detail)
+
+# Sanity Check: all funds for which a factset_fund_id exists has complete
+# coverage on reported_mv (complete in the sense that every security has an
+# associated market value, doesn't necessarily mean each fund has comprehensive
+# coverage of all securities)
+data %>%
+  mutate(factset_fund_id_exists = !is.na(factset_fund_id)) %>%
+  filter(factset_fund_id_exists) %>%
+  mutate(mv_exists = !is.na(reported_mv)) %>%
+  group_by(factset_fund_id_exists) %>%
+  summarize(isin_count = n())
+
+# Using the following two issues and asset maps, we can get info on what
+# asset_class is associated with each holding
+ref_v2_issue_type_map <- tbl(factset_db, "ref_v2_issue_type_map")
+ref_v2_asset_class_map <- tbl(factset_db, "ref_v2_asset_class_map")
+
+issue_type_code_asset_class_desc <- ref_v2_issue_type_map %>%
+  left_join(ref_v2_asset_class_map, by = "asset_class_code") %>%
+  select(issue_type_code, asset_class_desc)
+
+fsym_id_asset_type <- tbl(factset_db, "own_v5_own_sec_coverage") %>%
+  left_join(
+    issue_type_code_asset_class_desc,
+    by = c("issue_type" = "issue_type_code")
+    ) %>%
+  select(fsym_id, asset_type = asset_class_desc) %>%
+  collect()
+
+# connect asset type to fund data
+data <- data %>%
+  left_join(fsym_id_asset_type) %>%
+  select(fund_isin, reported_mv, asset_type, fsym_id)
+
+# connect ISINs to the fsym_ids containing each
+data <- data %>%
+  left_join(sym_isin) %>%
+  select(fund_isin, holding_isin = isin, reported_mv, asset_type)
+
+# It's important to note that each fund may have different types of holdings:
+# Equity, Fixed Income, Private Equity, Cash, Other
+# We don't expect Private Equity, Cash or Other to necessarily have corresponding ISINs
+# However, the Fixed Income class could be an issue.
+
+# Let's look at coverage by asset_type
+data %>%
+  filter(!is.na(asset_type)) %>%
+  mutate(holding_isin_exists = !is.na(holding_isin)) %>%
+  group_by(fund_isin, asset_type, holding_isin_exists) %>%
+  summarise(reported_mv = sum(reported_mv)) %>%
```

Comment:
is this line adding together all the reported mv values cells per fund_id?
![image](https://user-images.githubusercontent.com/55149721/168106234-8d057efb-1367-4c90-ae88-4340cbaf9a5c.png)

Reply:
Exactly. Notice how I'm not actually assigning this to anything, I just wanted to print the output to console as a sanity check. It allows you to see, per fund and per asset_class, what is the approximate coverage in terms of ISINs. This is interesting, because we might see that most funds have 100% coverage for the "Equity" asset class, and 0% coverage for the "Cash" asset class, which is legit because these different classes may or may not have ISINs.